sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -343,7 +347,11 @@ func (n *ProcessSizedElementsAndRestrictions) ProcessElement(_ context.Context,
 		defer func() {
 			<-n.SU
 		}()
-		return n.PDo.processSingleWindow(mainIn)
+		continuation, processResult := n.PDo.processSingleWindow(mainIn)
+		if continuation != nil {
+			n.continuation = continuation
+		}
```

Comment:
The invoker code already ensured that the continuation is not nil, so why check here? Also, the continuation field would likely *always* be nil to start, so overriding it with a nil doesn't change anything.

I'd rather that it gets overriden all the time, since it avoids cross bundle contamination to begin with.

Reply:
That's fair.